agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -753,37 +749,32 @@ public Dirs(File kitDir, File workingDir, RemoteKitManager kitManager) {
     }
 
     static Optional<Dirs> discover(InstanceId instanceId, String hostName, Distribution distribution, License license, String kitInstallationName, String kitInstallationPath) {
+      File kitPath = null;
       if (kitInstallationPath == null) {
         RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
         if (!kitManager.isKitAvailable()) {
           return Optional.empty();
         }
-
+        kitPath = kitManager.installKit(license, Collections.singletonList(hostName));
         logger.info("Installing kit on host {} from {}", hostName, distribution);
-        return Optional.of(new Dirs(
-            kitManager.installKit(license, Collections.singletonList(hostName)),
-            kitManager.getWorkingDir().toFile(),
-            kitManager
-        ));
-
       } else {
         // DO NOT ALTER THE KIT CONTENT IF kitInstallationPath IS USED
-        File kitDir = new File(kitInstallationPath);
-        Path workingPath = Agent.WORK_DIR.resolve(instanceId.toString());
-        try {
-          Files.createDirectories(workingPath);
-          if (license != null) {
-            license.writeToFile(workingPath.toFile());
-          }
-        } catch (IOException e) {
-          logger.debug("Can not create {}", workingPath, e);
+        kitPath = new File(kitInstallationPath);
+      }
+      Path workingPath = Agent.WORK_DIR.resolve(instanceId.toString());
+      try {
+        Files.createDirectories(workingPath);
+        if (license != null) {
+          license.writeToFile(workingPath.toFile());
         }
-        return Optional.of(new Dirs(
-            kitDir,
-            workingPath.toFile(),
-            null
-        ));
+      } catch (IOException e) {
+        logger.debug("Can not create {}", workingPath, e);
```

Comment:
while we are here... should we thrown an UncheckedIOException here is the license we want to install is not writable ? Otherwise we could end up running a test with a kit incorrectly installed

Reply:
done